bitcoin/src/address.rs

```diff
@@ -92,11 +94,11 @@ pub enum Error {
     NetworkValidation {
         /// Network that was required.
         required: Network,
-        /// Network on which the address was found to be valid.
-        found: Network,
         /// The address itself
-        address: Address<NetworkUnchecked>,
+        address: Box<Address<NetworkUnchecked>>,
```

Comment:
We should have this opaque and the internal representation (whether it's `Box` or not) should be bounded on `alloc`.

Reply:
No worries, I didn't put that much thought into the error stuff because I knew this was going to be draft, just boxed to keep clippy happy. In future I'll try document such things in the PR description to save you clock cycles during review.

Reply:
I knew why you did that, the lint is kinda stupid because it doesn't know we might want to have it no-alloc.

Reply:
What does the linter say? I don't see any reason why this should be boxed..

Reply:
```
warning: the `Err`-variant returned from this function is very large
   --> bitcoin/src/address.rs:412:59
    |
94  | /     NetworkValidation {
95  | |         /// Network that was required.
96  | |         required: Network,
97  | |         /// The address itself
98  | |         address: Address<NetworkUnchecked>,
99  | |     },
    | |_____- the largest variant contains at least 129 bytes
...
412 |       pub fn new<P>(version: WitnessVersion, program: P) -> Result<Self, Error>
    |                                                             ^^^^^^^^^^^^^^^^^^^
    |
    = help: try reducing the size of `address::Error`, for example by boxing large elements or replacing it with `Box<address::Error>`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#result_large_err
    = note: `#[warn(clippy::result_large_err)]` on by default
```